### PR TITLE
runtime_vm: use containerd deps for container io directly

### PR DESCRIPTION
**Pull request by gozssky**
#### What type of PR is this?

/kind dependency-change

#### What this PR does / why we need it:

Historically, a long time ago, a bunch of containerd code was brought into CRI-O to avoid vendoring. However, those code cannot  receive bug fixes from containerd.

See https://github.com/cri-o/cri-o/pull/5574#discussion_r801561012 for more details.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by fidencio** [comment]
I will go through this PR Tomorrow.  It's on my TODO list.

**Comment by haircommander** [comment]
/approve

@gozssky thanks for the PR! can you rebase to get test fixes

**Comment by gozssky** [comment]
@haircommander It seems all tests have passed.

**Comment by haircommander** [comment]
ping @fidencio @fgiudici for final lgtm

File: internal/oci/runtime_vm.go [hunk 1]
```diff
@@ -944,7 +944,8 @@ func (r *runtimeVM) AttachContainer(ctx context.Context, c *Container, inputStre
 		},
 	}
 
-	return cInfo.cio.Attach(opts)
+	cInfo.cio.Attach(opts)
+	return nil
```
Comment (fgiudici) [inline internal/oci/runtime_vm.go:948]: just a note for reviewers: we previously generated an error only if utils.GenerateID() failed (just the generation of a unique ID).
Also if we encountered an error while performing the Attach(), we logged it but returned nil.
So, looks fine to me always returning nil (as containerd's Attach() doesn't return any error).

**Review by fgiudici — COMMENTED** [review]
LGTM, thanks @gozssky :-)

**Comment by fidencio** [comment]
/lgtm

Thanks @gozssky for following up on this one, and sorry it took longer than expected for a review from my side.

**Pull request merged**